program/src/error.rs

```diff
@@ -88,6 +88,10 @@ pub enum StakeError {
     /// 19 - Invalid slash amount
     #[error("Invalid slash amount")]
     InvalidSlashAmount,
+
+    /// 20 - Undelegated stake state account
+    #[error("Undelegated stake state account")]
+    UndelegatedStakeStateAccount,
```

Comment:
nit: how about calling it a "SOL stake account"?
```suggestion
    /// 20 - Undelegated SOL stake account
    #[error("Undelegated SOL stake account")]
    UndelegatedSolStakeAccount,
```